src/xclim/core/calendar.py

```diff
@@ -1629,6 +1630,18 @@ def unstack_periods(da: xr.DataArray | xr.Dataset, dim: str = "period") -> xr.Da
             f"`unstack_periods` can't find the window, stride and freq attributes on the {dim} coordinates."
         ) from err
 
+    src_freq = xr.infer_freq(da.time)
+    # Ok freqs are < D and uniform-calendar >= Y.
+    if not (
+        compare_offsets(src_freq, "<", "MS")
+        or (compare_offsets(src_freq, ">=", "YS") and da.time.dt.calendar in uniform_calendars)
+    ):
+        warnings.warn(
+            "xclim is not able to unstack periods that were constructed from non-uniform time steps. "
+            f"(Found : {src_freq} with calendar {da.time.dt.calendar}). "
+            "Results won't fit with the original coordinates passed to `stack_periods`."
```

Comment:
This is a nitpick, but it's a good idea to not have string formatting directly in logging calls.

I tend to employ `msg = f"{object}"; warnings.warn(msg)`.